### PR TITLE
Added storybook to signup form

### PR DESCRIPTION
**Pull request by binary-koan**
refs https://github.com/TryGhost/Team/issues/3299

This uses the standard Storybook layout I'm familiar with - not sure how to set up the alternative layout from Koenig, and not sure if that's something we prefer to use in general.

### CODE REVIEW

File: ghost/signup-form/.vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      "editorGroupHeader.tabsBackground": "#1b1e2b",
+      "tab.border": "#00000000",
+      "sideBarSectionHeader.border": "#00000000",
+      "terminal.border": "#a6accd"
+    },
+    "activityBar.background": "#540E48",
+    "titleBar.activeBackground": "#761365",
+    "titleBar.activeForeground": "#FEFCFE"
+  }
+}
```
Comment (SimonBackx) [inline ghost/signup-form/.vscode/settings.json:17]: Should probably be added to gitignore

Comment (binary-koan) [reply ghost/signup-form/.vscode/settings.json]: Oops, that was just from me opening the repo outside a workspace one time. I'll just delete it, shouldn't be added normally

**Review by SimonBackx — APPROVED** [review]
Just one comment is that I'm not sure if we should keep FormView and SuccessView + their stories inside the pages folder as they aren't a page any longer. Maybe they could also stay within the same file (FormPage) to keep it easier to edit?

**Comment by binary-koan** [comment]
@SimonBackx Hmm, not sure - it also feels a bit weird moving them into the same files and exporting them as separate components purely for storybook. I would maybe just put everything under `components` rather than having a `pages` directory at all since the total number of components is still small, and figure out a better structure once it becomes hard to manage

I'll merge this as is to unblock the other tasks, but feel free to reorganise the components later!

**Pull request merged**